Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -269,12 +269,16 @@ final class PurchasesOrchestrator {
                           product: StoreProductType,
                           completion: @escaping @Sendable (Result<PromotionalOffer, PurchasesError>) -> Void) {
         guard let discountIdentifier = productDiscount.offerIdentifier else {
-            completion(.failure(ErrorUtils.productDiscountMissingIdentifierError()))
+            self.operationDispatcher.dispatchOnMainActor {
+                completion(.failure(ErrorUtils.productDiscountMissingIdentifierError()))
+            }
             return
         }
 
         guard let subscriptionGroupIdentifier = product.subscriptionGroupIdentifier else {
-            completion(.failure(ErrorUtils.productDiscountMissingSubscriptionGroupIdentifierError()))
+            self.operationDispatcher.dispatchOnMainActor {
+                completion(.failure(ErrorUtils.productDiscountMissingSubscriptionGroupIdentifierError()))
+            }
```

Comment:
guess these were relying on the fact that no async operations have taken place yet so it'd have just called completion from the calling thread

Reply:
Yep exactly